networkx/algorithms/cluster.py

```diff
@@ -254,6 +254,63 @@ def wt(u, v):
         yield (i, dtotal, dbidirectional, float(directed_triangles))
 
 
+@not_implemented_for("directed")
+@nx._dispatchable
+def all_triangles(G, nodes=None):
+    """
+    Yields all unique triangles in an undirected graph using a node ordering strategy.
+
+    A triangle is a set of three distinct nodes where each node is connected to the other two.
+
+    Parameters
+    ----------
+    G : NetworkX graph
+        An undirected graph. Self-loops and multiple edges are ignored.
```

Comment:
Minor nit: I believe this doesn't matter (self-loops and multiedges shouldn't affect the result), and it might just end up confusing users ("will this function work for multigraphs?").
```suggestion
        An undirected graph.
```